core/src/main/scala/clue/websocket/package.scala

```diff
@@ -3,25 +3,32 @@
 
 package clue
 
+import cats.syntax.all._
+
+import scala.concurrent.duration.FiniteDuration
+
 package object websocket {
-  type WebSocketReconnectionStrategy = ReconnectionStrategy[WebSocketCloseEvent]
+  type CloseEvent = Either[Throwable, CloseParams]
 
-  object WebSocketReconnectionStrategy {
-    def never: WebSocketReconnectionStrategy = ReconnectionStrategy.never
-  }
+  type CloseReason          = Either[Throwable, CloseEvent]
```

Comment:
What is the difference between `Right(Left(ex))` and `Left(ex)` ?

Reply:
I added comments in the code. In short, one is a "lost connection" exception, the other one is an "establishing connection" exception.

```scala
  // A Left(exception) here indicates an unexpected/errored disconnection from the server.
  type CloseEvent = Either[Throwable, CloseParams]

  // A Left(exception) here indicates an error establishing the connection.
  type ReconnectReason = Either[Throwable, CloseEvent]

  // Int = Attempt #. Will only be 0 immediately after a close (whether graceful or errored).
  // If the very fist connection attempt fails, the first invocation will have Attempt = 1.
  type ReconnectionStrategy = (Int, ReconnectReason) => Option[FiniteDuration]
```

Reply:
Also renamed `CloseReason` to the more adequate `ReconnectReason`.

Reply:
Thanks, that helps!